CommonTools/PileupAlgos/src/PuppiContainer.cc

```diff
@@ -130,51 +137,55 @@ void PuppiContainer::getRMSAvg(int iOpt,std::vector<PuppiCandidate> const &iCons
         int  pAlgo    = fPuppiAlgo[pPupId].algoId   (iOpt);
         bool pCharged = fPuppiAlgo[pPupId].isCharged(iOpt);
         double pCone  = fPuppiAlgo[pPupId].coneSize (iOpt);
-        //Compute the Puppi Metric
-        if(!pCharged) pVal = goodVar(iConstits[i0],iParticles       ,pAlgo,pCone);
-        if( pCharged) pVal = goodVar(iConstits[i0],iChargedParticles,pAlgo,pCone);
+        // compute the Puppi metric:
+        //  - calculate goodVar only for candidates that (1) will not be assigned a predefined weight (e.g 0, 1),
+        //    or (2) are required for computations inside puppi-algos (see call to PuppiAlgo::add below)
+        double pVal = -1;
+        bool const getsDefaultWgtIfApplyCHS = iConstits[i0].id == 1 or iConstits[i0].id == 2;
+        if (not(fApplyCHS and getsDefaultWgtIfApplyCHS) or (std::abs(iConstits[i0].eta) < fPuppiAlgo[pPupId].etaMaxExtrap() and iConstits[i0].puppi_register != 0)) {
```

Comment:
The master version uses `and getsDefaultWgtIfApplyCHS` in the second conditional instead of `and iConstits[i0].puppi_register != 0`. Are these equivalent? If yes, a redefinition of `getsDefaultWgtIfApplyCHS` may be in order.
If no, please elaborate
(in this case, the separate definition of `getsDefaultWgtIfApplyCHS` seems unnecessary because it is not reused anywhere else).

also a linebreak at/after `or` would be nice

Reply:
In practice, I think the condition `id == 1 or id == 2` corresponds to `register != 0` (looking at how `puppi_register` is filled [here](https://github.com/cms-sw/cmssw/pull/31290/files#diff-920ad87cd21790b241508fda9a732044L56)), but in principle the two things can differ (or at least, I wanted to make no assumptions).

The logic behind how these changes are implemented is the following: `goodVar` needs to be calculated when the final weight of the candidate is non-trivial (this is the first condition, which corresponds to the logical NOT of [these two cases](https://github.com/cms-sw/cmssw/blob/a27570aa535de693b718c4abd25258f6dfb694fc/CommonTools/PileupAlgos/src/PuppiContainer.cc#L275-L276)), or when the `goodVar` of a given candidate is needed internally by [`PuppiAlgo::add`, here](https://github.com/cms-sw/cmssw/blob/a27570aa535de693b718c4abd25258f6dfb694fc/CommonTools/PileupAlgos/src/PuppiAlgo.cc#L110-L116); in `10_6_X`, the latter condition translates to `(.. < etaMaxExtrap) and puppi_register != 0`, while in `11_X_Y` the criterion is `(.. < etaMaxExtrap) and (id == 1 or id == 2)` (see [here](https://github.com/cms-sw/cmssw/blob/af81ead1e2eecda9b39833574cd462337628d084/CommonTools/PileupAlgos/src/PuppiAlgo.cc#L100-L106)).

So, the logic is the same here and in the original PR; the end result looks different just as a consequence of the fact that the `Puppi{Container,Algo}` implementation is different in the two releases.

I added a commit to remove the temporary variable `getsDefaultWgtIfApplyCHS`, which is indeed only used once, and took the opportunity to add the linebreak.